Common/include/toolboxes/multilayer_perceptron/CIOMap.hpp

```diff
@@ -0,0 +1,190 @@
+/*!
+ * \file CIOMap.hpp
+ * \brief Declaration of input-to-output mapping class for artifical neural networks
+ * \author E. Bunschoten
+ * \version 7.5.0 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2022, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include <cmath>
+#include <cstdlib>
+#include <iostream>
+#include <limits>
+#include <string>
+#include <vector>
+
+#include "../../linear_algebra/blas_structure.hpp"
+#include "CLookUp_ANN.hpp"
+
+namespace MLPToolbox {
+
+class CLookUp_ANN;
+
+class CIOMap
+/*!
+ *\class CIOMap
+ *\brief This class is used by the CLookUp_ANN class to assign user-defined inputs
```

Comment:
Documentation above declaration, please add a doxygen group in docmain.hpp and place these classes in that group.

Reply:
I made the MLP classes indeed as general as possible, such that they could be used in a variety of applications within SU2 (fluid models, sub-grid scale models, etc). Apart from MPI, variable types like su2double, MapType, and su2vector are used. I have not considered hosing the MLP as a personal repo. What would be the benefit of this?

Reply:
> Documentation above declaration, please add a doxygen group in docmain.hpp and place these classes in that group.

What do you mean exactly by "documentation above declaration"?

Reply:
"Documentation"
class Class {
...

Reply:
Less non-CFD code for us to review.
Easier for you or someone else to re-use in other projects.
The types are easy to solve if you make it header-only, the user can typedef before including the main header. Or you can template the classes.